### PR TITLE
Standardize ByT5 model card format

### PR DESCRIPTION
**Pull request by yanamis**
# What does this PR do?

This PR updates the ByT5 model card to follow the standardized format as requested in #36979. 

## Before submitting
- [x] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#create-a-pull-request),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

@stevhliu

### CODE REVIEW

File: docs/source/en/model_doc/byt5.md [hunk 1]
```diff
@@ -13,150 +13,109 @@ specific language governing permissions and limitations under the License.
...
-">
-</div>
+[ByT5: Towards a token-free future with pre-trained byte-to-byte models](https://arxiv.org/abs/2105.13626) was proposed by Linting Xue, Aditya Barua, Noah Constant, Rami Al-Rfou, Sharan Narang, Mihir Kale, Adam Roberts, and Colin Raffel.
```
Comment (stevhliu) [inline docs/source/en/model_doc/byt5.md]:
```suggestion
```

File: docs/source/en/model_doc/byt5.md [hunk 2]
```diff
@@ -13,150 +13,109 @@ specific language governing permissions and limitations under the License.
...
 
-## Overview
+ByT5 is a version of the T5 model that works directly on **raw UTF-8 bytes** instead of using tokens or subwords. That means no tokenizer is needed — it just takes raw text, byte by byte. This makes it naturally multilingual, more robust to spelling/noise, and simpler to use, since there's no complex text preprocessing.
 
-The ByT5 model was presented in [ByT5: Towards a token-free future with pre-trained byte-to-byte models](https://arxiv.org/abs/2105.13626) by Linting Xue, Aditya Barua, Noah Constant, Rami Al-Rfou, Sharan Narang, Mihir
-Kale, Adam Roberts, Colin Raffel.
+You can find all the original ByT5 checkpoints under the [ByT5 model collection](https://huggingface.co/google).
```
Comment (stevhliu) [inline docs/source/en/model_doc/byt5.md]:
```suggestion
You can find all the original ByT5 checkpoints under the [Google](https://huggingface.co/google?search_models=byt5) organization.
```

File: docs/source/en/model_doc/byt5.md [hunk 3]
```diff
@@ -13,150 +13,109 @@ specific language governing permissions and limitations under the License.
...
 
-The abstract from the paper is the following:
+> [!TIP]
+> Click on the ByT5 models in the right sidebar for more examples of how to apply ByT5 to different text generation and translation tasks.
```
Comment (stevhliu) [inline docs/source/en/model_doc/byt5.md]:
```suggestion
> Refer to the [T5](./t5) docs for more examples of how to apply ByT5 to different language tasks..
```

File: docs/source/en/model_doc/byt5.md [hunk 4]
```diff
@@ -13,150 +13,109 @@ specific language governing permissions and limitations under the License.
...
-">
-</div>
+[ByT5: Towards a token-free future with pre-trained byte-to-byte models](https://arxiv.org/abs/2105.13626) was proposed by Linting Xue, Aditya Barua, Noah Constant, Rami Al-Rfou, Sharan Narang, Mihir Kale, Adam Roberts, and Colin Raffel.
 
-## Overview
+ByT5 is a version of the T5 model that works directly on **raw UTF-8 bytes** instead of using tokens or subwords. That means no tokenizer is needed — it just takes raw text, byte by byte. This makes it naturally multilingual, more robust to spelling/noise, and simpler to use, since there's no complex text preprocessing.
```
Comment (stevhliu) [inline docs/source/en/model_doc/byt5.md]:
```suggestion
[ByT5](https://huggingface.co/papers/2105.13626) is tokenizer-free version of the [T5](./t5) model designed to works directly on raw UTF-8 bytes. This means it can process any language, more robust to noise like typos, and simpler to use because it doesn't require a preprocessing pipeline.
```

File: docs/source/en/model_doc/byt5.md [hunk 5]
```diff
@@ -13,150 +13,109 @@ specific language governing permissions and limitations under the License.
...
-removing complex and error-prone text preprocessing pipelines. Since byte or character sequences are longer than token
-sequences, past work on token-free models has often introduced new model architectures designed to amortize the cost of
-operating directly on raw text. In this paper, we show that a standard Transformer architecture can be used with
-minimal modifications to process byte sequences. We carefully characterize the trade-offs in terms of parameter count,
-training FLOPs, and inference speed, and show that byte-level models are competitive with their token-level
-counterparts. We also demonstrate that byte-level models are significantly more robust to noise and perform better on
-tasks that are sensitive to spelling and pronunciation. As part of our contribution, we release a new set of
-pre-trained byte-level Transformer models based on the T5 architecture, as well as all code and data used in our
-experiments.*
+The example below demonstrates how to run ByT5 using either [`Pipeline`] or the [`AutoModel`] class.
```
Comment (stevhliu) [inline docs/source/en/model_doc/byt5.md]:
```suggestion
The example below demonstrates how to generate text with [`Pipeline`], [`AutoModel`] and from the command line.
```

File: docs/source/en/model_doc/byt5.md [hunk 6]
```diff
@@ -13,150 +13,109 @@ specific language governing permissions and limitations under the License.
...
 
-This model was contributed by [patrickvonplaten](https://huggingface.co/patrickvonplaten). The original code can be
-found [here](https://github.com/google-research/byt5).
+<hfoptions id="usage">
+<hfoption id="Pipeline">
 
-<Tip>
+```python
+from transformers import pipeline
```
Comment (stevhliu) [inline docs/source/en/model_doc/byt5.md:40]:
```py
import torch
from transformers import pipeline

pipeline = pipeline(
    task="text2text-generation",
    model="google/byt5-small",
    torch_dtype=torch.float16,
    device=0
)
pipeline("translate English to French: The weather is nice today")
```

File: docs/source/en/model_doc/byt5.md [hunk 7]
```diff
@@ -13,150 +13,109 @@ specific language governing permissions and limitations under the License.
...
+inputs = tokenizer(["Life is like a box of chocolates.", "Today is Monday."], padding="longest", return_tensors="pt")
+labels = tokenizer(["La vie est comme une boîte de chocolat.", "Aujourd'hui c'est lundi."], padding="longest", return_tensors="pt").input_ids
+
+loss = model(**inputs, labels=labels).loss
+print(loss.item())
+```
 
->>> input_ids = torch.tensor([list("Life is like a box of chocolates.".encode("utf-8"))]) + num_special_tokens
+</hfoption>
+</hfoptions>
```
Comment (stevhliu) [inline docs/source/en/model_doc/byt5.md:81]: Include an example from the command line

File: docs/source/en/model_doc/byt5.md [hunk 8]
```diff
@@ -13,150 +13,109 @@ specific language governing permissions and limitations under the License.
...
 
->>> labels = torch.tensor([list("La vie est comme une boîte de chocolat.".encode("utf-8"))]) + num_special_tokens
+## Quantization
 
->>> loss = model(input_ids, labels=labels).loss
->>> loss.item()
-2.66
+Quantization reduces the memory burden of large models by using lower-precision weights. While ByT5 comes in relatively small versions (`small`, `base`, etc.), quantization can still be applied using libraries like `bitsandbytes` or `optimum`. For more on this, refer to the [Quantization overview](../quantization/overview).
```
Comment (stevhliu) [inline docs/source/en/model_doc/byt5.md]:
```suggestion
Quantization reduces the memory burden of large models by representing the weights in a lower precision. Refer to the [Quantization](../quantization/overview) overview for more available quantization backends.

The example below uses [torchao](../quantization/torchao) to only quantize the weights to int4.
```

File: docs/source/en/model_doc/byt5.md [hunk 9]
```diff
@@ -13,150 +13,109 @@ specific language governing permissions and limitations under the License.
...
+
+Example:
+```python
+# Use with 8-bit loading
+from transformers import AutoModelForSeq2SeqLM
+
+model = AutoModelForSeq2SeqLM.from_pretrained("google/byt5-small", load_in_8bit=True, device_map="auto")
```
Comment (stevhliu) [inline docs/source/en/model_doc/byt5.md]:
```suggestion
# pip install torchao
import torch
from transformers import TorchAoConfig, AutoModelForSeq2SeqLM, AutoTokenizer

quantization_config = TorchAoConfig("int4_weight_only", group_size=128)
model = AutoModelForSeq2SeqLM.from_pretrained(
    "google/byt5-xl",
    torch_dtype=torch.bfloat16,
    device_map="auto",
    quantization_config=quantization_config
)

tokenizer = AutoTokenizer.from_pretrained("google/byt5-xl")
input_ids = tokenizer("translate English to French: The weather is nice today.", return_tensors="pt").to("cuda")

output = model.generate(**input_ids)
print(tokenizer.decode(output[0], skip_special_tokens=True))
```

File: docs/source/en/model_doc/byt5.md [hunk 10]
```diff
@@ -13,150 +13,109 @@ specific language governing permissions and limitations under the License.
...
-ByT5's architecture is based on the T5v1.1 model, refer to [T5v1.1's documentation page](t5v1.1) for the API reference. They
-only differ in how inputs should be prepared for the model, see the code examples below.
+pipe = pipeline("text2text-generation", model="google/byt5-small")
+result = pipe("Translate English to French: Life is beautiful.")
+print(result)
+```
 
-</Tip>
+</hfoption>
+<hfoption id="AutoModel">
```
Comment (stevhliu) [inline docs/source/en/model_doc/byt5.md:52]: Let's only show the tokenizer example here and we can make a note of how to use it without a tokenizer in the `## Notes` section

```py
import torch
from transformers import AutoModelForSeq2SeqLM, AutoTokenizer

tokenizer = AutoTokenizer.from_pretrained(
    "google/byt5-small"
)
model = AutoModelForSeq2SeqLM.from_pretrained(
    "google/byt5-small",
    torch_dtype=torch.float16,
    device_map="auto"
)

input_ids = tokenizer("summarize: Photosynthesis is the process by which plants, algae, and some bacteria convert light energy into chemical energy.", return_tensors="pt").to("cuda")

output = model.generate(**input_ids)
print(tokenizer.decode(output[0], skip_special_tokens=True))
```

File: docs/source/en/model_doc/byt5.md [hunk 11]
```diff
@@ -13,150 +13,109 @@ specific language governing permissions and limitations under the License.
...
+Example:
+```python
+# Use with 8-bit loading
+from transformers import AutoModelForSeq2SeqLM
+
+model = AutoModelForSeq2SeqLM.from_pretrained("google/byt5-small", load_in_8bit=True, device_map="auto")
 ```
 
-For batched inference and training it is however recommended to make use of the tokenizer:
+## Attention Visualization
```
Comment (stevhliu) [inline docs/source/en/model_doc/byt5.md]: I don't think its implemented for T5 yet so you can remove this

File: docs/source/en/model_doc/byt5.md [hunk 12]
```diff
@@ -13,150 +13,109 @@ specific language governing permissions and limitations under the License.
...
 
-Similar to [T5](t5), ByT5 was trained on the span-mask denoising task. However, 
-since the model works directly on characters, the pretraining task is a bit 
-different. Let's corrupt some characters of the 
-input sentence `"The dog chases a ball in the park."` and ask ByT5 to predict them 
-for us.
+## Notes
+
+- ByT5 uses **no traditional tokenizer**. Instead, it operates directly on UTF-8 bytes. That means the input must be properly encoded, and users must account for the three special tokens (IDs 0, 1, 2).
+- Masking is also unique: instead of sentinel tokens like `{extra_id_0}`, ByT5 uses the top byte values (e.g. 258, 257...) for masking.
```
Comment (stevhliu) [inline docs/source/en/model_doc/byt5.md]:
```suggestion
- It is recommended to use the tokenizer for batched inference and training.
- The example below shows how to use the model without a tokenizer.

   ```py
   import torch
   from transformers import AutoModelForSeq2SeqLM
   
   model = AutoModelForSeq2SeqLM.from_pretrained("google/byt5-small")
   
   num_special_tokens=3
   input_ids = torch.tensor([list("Life is like a box of chocolates.".encode("utf-8"))]) + num_special_tokens
   labels = torch.tensor([list("La vie est comme une boîte de chocolat.".encode("utf-8"))]) + num_special_tokens
   loss = model(input_ids, labels=labels).loss
   loss.item()
   ```
```

File: docs/source/en/model_doc/byt5.md [hunk 13]
```diff
@@ -13,150 +13,109 @@ specific language governing permissions and limitations under the License.
...
->>> output_string = tokenizer.batch_decode(output_ids_list)
->>> output_string
-['<pad>', 'is the one who does', ' in the disco', 'in the park. The dog is the one who does a ball in', ' in the park.']
+# Example: character-level denoising with mask tokens
+input_ids = tokenizer("The dog chases a ball in the park.").input_ids
+masked_input = torch.tensor([input_ids[:8] + [258] + input_ids[14:21] + [257] + input_ids[28:]])
+output = model.generate(masked_input, max_length=100)
 ```
 
+ByT5 was contributed by [patrickvonplaten](https://huggingface.co/patrickvonplaten). The original implementation is available on [GitHub](https://github.com/google-research/byt5).
```
Comment (stevhliu) [inline docs/source/en/model_doc/byt5.md]:
```suggestion
```

**Review by stevhliu — COMMENTED** [review]
Thanks for contributing!

**Comment by yanamis** [comment]
Thanks @stevhliu for the detailed feedback! I've updated the model card as suggested. Please let me know if anything else needs adjusting.

File: docs/source/en/model_doc/byt5.md [hunk 14]
```diff
@@ -13,150 +13,128 @@ specific language governing permissions and limitations under the License.
...
+print(tokenizer.decode(output[0], skip_special_tokens=True))
 ```
 
-For batched inference and training it is however recommended to make use of the tokenizer:
+## Notes
+
+- It is recommended to use the tokenizer for batched inference and training.
+- The example below shows how to use the model without a tokenizer.
 
 ```python
```
Comment (stevhliu) [inline docs/source/en/model_doc/byt5.md]: Indent the code block under the list item above ^

File: docs/source/en/model_doc/byt5.md [hunk 15]
```diff
@@ -13,150 +13,128 @@ specific language governing permissions and limitations under the License.
...
+loss = model(input_ids, labels=labels).loss
+loss.item()
 ```
 
-Similar to [T5](t5), ByT5 was trained on the span-mask denoising task. However, 
-since the model works directly on characters, the pretraining task is a bit 
-different. Let's corrupt some characters of the 
-input sentence `"The dog chases a ball in the park."` and ask ByT5 to predict them 
-for us.
+- Masking is also unique: instead of sentinel tokens like `{extra_id_0}`, ByT5 uses the top byte values (e.g. 258, 257...) for masking.
```
Comment (stevhliu) [inline docs/source/en/model_doc/byt5.md]:
```suggestion
- ByT5 uses the top byte values (258, 257, etc.) for masking instead of sentinel tokens like `{extra_id_0}`.
```

File: docs/source/en/model_doc/byt5.md [hunk 16]
```diff
@@ -13,150 +13,128 @@ specific language governing permissions and limitations under the License.
...
 
 ```python
```
Comment (stevhliu) [inline docs/source/en/model_doc/byt5.md]: Also indent this code block under the list item above ^

**Review by stevhliu — APPROVED** [review]
Just a few more comments and then we can merge :)

**Comment by yanamis** [comment]
Everything's done now. Thanks again for your help :)

**Comment by stevhliu** [comment]
Can you try running `make repo-consistency` from the command line to fix the `check_repository_consistency` job?

**Comment by yanamis** [comment]
I've rebased the branch to bring it up to date and pushed the fix for the repo check error. Please let me know if it's all good now :)
